invenio_rdm_records/contrib/journal/custom_fields.py

```diff
@@ -12,14 +12,181 @@
 - journal.volume
 """
 
+from abc import ABC
+
 from idutils import is_issn
 from invenio_i18n import lazy_gettext as _
 from invenio_records_resources.services.custom_fields import BaseCF
-from invenio_records_resources.services.records.facets import CFTermsFacet
-from marshmallow import fields
+from marshmallow import fields, missing
 from marshmallow_utils.fields import SanitizedUnicode
 
 
+class FieldDumper(ABC):
+    """Mixin class that defines pre_dump and post_dump methods to extend records serialization."""
+
+    def post_dump(self, data, original={}, **kwargs):
+        """Hook called after the marshmallow serialization of the record.
+
+        :param data: The dumped record data.
+        :param original: The original record data.
+        :param kwargs: Additional keyword arguments.
+        :returns: The serialized record data.
+        """
+        return data
+
+    def pre_dump(self, data, original={}, **kwargs):
+        """
+        Hook called before the marshmallow serialization of the record.
+
+        :param data: The record data to dump.
+        :param original: The original record data.
+        :param kwargs: Additional keyword arguments.
+        :returns: The data to dump.
+        """
+        return data
+
+
+class FieldLoader(ABC):
+    def post_load(data, many=False, **kwargs):
+        raise NotImplementedError()
+
+    def pre_load(data, many=False, **kwargs):
+        raise NotImplementedError()
+
+
+class JournalDataciteDumper(FieldDumper):
+    """Dump processor for datacite serialization of 'Journal' custom field."""
+
+    def post_dump(self, data, original={}, **kwargs):
+        """Adds the journal information as a related identifier in the DataCite metadata."""
+        custom_fields = original.get("custom_fields", {})
```

Comment:
```suggestion
        custom_fields = (original or {}).get("custom_fields", {})
```
This would enable you to set `original=None` in the method's argument list above.